flexget/plugins/sites/lostfilm.py

```diff
@@ -47,147 +37,128 @@ class LostFilm(object):
-                title_eng_div = page.find(
-                    'div', 'inner-box--subtitle').text.strip() or None
-                series_name_eng = (title_eng_div.endswith(
-                    ', сериал')) and title_eng_div[:-8] or None
-                text_div = page.find(
-                    'div', 'inner-box--text').text.strip() or None
-                episode_name_rus, episode_name_eng = TEXT_REGEXP.findall(
-                    text_div).pop()
+                index_page = get_soup(response.content)
             except:
-                log.debug('Cannot parse head info')
-                pass
+                raise PluginError('Cannot parse new releases page')
+            # Iterate over releases
+            seen_buttons = index_page.find_all('div', 'haveseen-btn')
+            for seen_btn in seen_buttons:
+                try:
+                    details_pane = seen_btn.parent.find('div', 'details-pane')
+
+                    episode_name_rus = details_pane.find('div', 'alpha').text.strip()
+                    episode_name_eng = details_pane.find('div', 'beta').text.strip()
+                    series_name_rus = details_pane.parent.find('div', 'name-ru').text.strip()
+                    series_name_eng = details_pane.parent.find('div', 'name-en').text.strip()
+                    lostfilm_num, season_num, episode_num = [int(x) for x in seen_btn['data-code'].split('-')]
+                except AttributeError:
+                    log.exception('Could not parse {:s}, probably HTML markup was changed'.format(release_url))
+                    continue
 
-            episode_id = 'S{:02d}E{:02d}'.format(
-                season_num, episode_num)
+                params = {'c': lostfilm_num, 's': season_num, 'e': episode_num}
+                redirect_url = V_SEARCH_URL
+                try:
+                    response = task.requests.get(redirect_url, params=params)
+                except RequestException as e:
+                    log.error('Could not connect to redirect url: {:s}'.format(e))
+                    continue
 
-            for item in page.findAll('div', 'inner-box--item'):
-                torrent_link = quality = None
+                page = get_soup(response.content)
                 try:
-                    torrent_link = item.find(
-                        'div', 'inner-box--link sub').a['href']
-                    quality = item.find('div', 'inner-box--label').text.strip()
+                    redirect_url = page.head.meta['content'].split('url=')[1]  # retre.org
                 except:
-                    log.debug('Item doesn\'t have a link or quality')
+                    log.error('Missing redirect')
                     continue
-                if torrent_link is None or quality is None:
-                    log.debug('Item doesn\'t have a link or quality')
+
+                try:
+                    response = task.requests.get(redirect_url)
```

Comment:
Why not just do this in url_rewrite? Seems like a lot of requests on the get-go.

Reply:
Can you please give me a link to the docs? I'm a Ruby guy mostly, and this is my second Python experience :blush:, so I appreciate Python coding advices.

Reply:
It's not Python specific. Just look at some of the other url_rewrite plugins. The idea is to only fetch the .torrent link when something is accepted.

Reply:
Hmm, you're talking about [this](https://flexget.com/URLRewriters), right? I need some more time to figure out how it works...